go/ql/test/library-tests/semmle/go/frameworks/gqlgen/graph/schema.resolvers.go

```diff
@@ -0,0 +1,30 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.34
+
+import (
+	"context"
+	"fmt"
+	"pwntester/gqlgen-todos/graph/model"
+)
+
+// CreateTodo is the resolver for the createTodo field.
+func (r *mutationResolver) CreateTodo(ctx context.Context, input model.NewTodo) (*model.Todo, error) { // $ resolverParameter="definition of input"
```

Comment:
@owen-mc Thanks! is there any documentation about the syntax of these tests? what is the `$` for? can you add any string instead of `definition of input`?

Reply:
Ok, seems like every line where `Gqlgen::ResolverParameter` matches will be tagged with the `resolverParameter` tag with its `toString` value assigned to the tag value, so in the inline test we can check for tags with specific values. I guess the `$` is just to flag that line as an inline test and not just an ordinary comment?

Reply:
The documentation is at the top of [InlineExpectationsTest.qll](https://github.com/github/codeql/blob/d57276ca35f6201ecfcbed16ecc55278982078eb/shared/util/codeql/util/test/InlineExpectationsTest.qll). `"definition of input"` is defined by `value` in `hasActualResult` in `go/ql/test/library-tests/semmle/go/frameworks/gqlgen/gqlgen.ql`. `definition of X` is the `toString()` of a `ParameterNode` for a parameter named `X`.

Reply:
Thanks! Will use these tests the next time

Reply:
I should also mention [InlineFlowTest](https://github.com/github/codeql/blob/d57276ca35f6201ecfcbed16ecc55278982078eb/go/ql/test/TestUtilities/InlineFlowTest.qll), which is very easy to use.